hummingbot/wallet/ethereum/watcher/websocket_watcher.py

```diff
@@ -0,0 +1,169 @@
+        }
+        nonce = await self._send(emit_data)
+        raw_message = await self._client.recv()
+        if raw_message is not None:
+            resp = ujson.loads(raw_message)
+            if resp.get("id", None) == nonce:
+                self._node_address = resp.get("result")
+                return True
+        return False
+
+    async def unsubscribe(self, params) -> bool:
+        emit_data = {
+            "method": "eth_unsubscribe",
+            "params": params
+        }
+        nonce = await self._send(emit_data)
+        raw_message = await self._client.recv()
+        resp = ujson.loads(raw_message)
+        if resp.get("id", None) == nonce:
+            result = resp.get("result", None)
+            return (result is not None) and result
+
+    async def fetch_new_blocks_loop(self):
+        try:
+            while True:
+                try:
+                    raw_message = await self._client.recv()
+                    message_json = ujson.loads(raw_message) if raw_message is not None else None
+                    if message_json.get("method", None) == "eth_subscription":
+                        subscription_result_params = message_json.get("params", None)
+                        incoming_block = subscription_result_params.get("result", None) \
+                            if subscription_result_params is not None else None
+                        if incoming_block is not None:
+                            new_block: AttributeDict = await self.call_async(self._w3.eth.getBlock,
+                                                                             incoming_block.get("hash"), True)
+                            self._current_block_number = new_block.get("number")
+                            self._block_cache[new_block.get("hash")] = new_block
+                            self.trigger_event(NewBlocksWatcherEvent.NewBlocks, [new_block])
+                except asyncio.CancelledError:
+                    if self._network_on:
+                        self.logger().network("Network closed abnormally, reconnecting ...")
+                        self.connect()
+                    raise
+                except asyncio.TimeoutError:
+                    self.logger().network("Timed out fetching new block.", exc_info=True,
+                                          app_warning_msg="Timed out fetching new block. "
+                                                          "Check wallet network connection")
+                except BlockNotFound:
+                    pass
+                except ConnectionClosedOK:
+                    raise
+                except Exception as e:
+                    self.logger().network(f"Error fetching new block: {e}", exc_info=True,
+                                          app_warning_msg="Error fetching new block. "
+                                                          "Check wallet network connection")
+        except asyncio.CancelledError:
```

Comment:
This except and raise here is unnecessary, you need to do this only when you also catch other kinds of Exceptions.

Reply:
You don't need to except asyncio.CancelledError here, but you do need one in your inner try-except block.